### PR TITLE
[14.0][ADD] sale_commission_agent_restrict

### PR DESCRIPTION
**Pull request by ilyasProgrammer**

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
Functional review ok!

**Review by elvise — APPROVED** [review]
LGTM!

**Comment by francesco-ooops** [comment]
@OCA/crm-sales-marketing-maintainers what do you think of this?

**Comment by francesco-ooops** [comment]
@OCA/crm-sales-marketing-maintainers any feedback? :)

File: sale_commission_agent_restrict/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from . import models
+from odoo import api, SUPERUSER_ID
+
+# -- List of predefined rules that must be managed
+PREDEFINED_RULES = [
```
Comment (rousseldenis) [inline sale_commission_agent_restrict/__init__.py]: It's better to add a hooks.py file at the module root in order to find it easily.

**Comment by elvise** [comment]
@ilyasProgrammer please check pre-commit and codecov

**Comment by ilyasProgrammer** [comment]
Do i need to add tests for uninstall hook? I don't know how it is done.

**Comment by ilyasProgrammer** [comment]
@rousseldenis @OCA/crm-sales-marketing-maintainers can be merged now ?

File: sale_commission_agent_restrict/models/sale.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    @api.depends("order_line.agent_ids.amount")
+    def _compute_commission_total(self):
+        for record in self:
+            record.commission_total = sum(record.mapped("order_line.agent_ids.amount"))
+            my_com = record.mapped("order_line.agent_ids").filtered(
+                lambda x: x.agent_id == self.env.user.partner_id
+            )
+            record.current_agent_total = sum(my_com.mapped("amount"))
+
+    current_agent_total = fields.Float(
```
Comment (rousseldenis) [inline sale_commission_agent_restrict/models/sale.py]: Please put fields on top

File: sale_commission_agent_restrict/security/security.xml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            ('id', 'child_of', user.partner_id.child_ids.ids),
+            ('user_ids', 'in', user.company_id.user_ids.ids),
+            ('agent_ids', 'in', user.partner_id.ids),
+            ('active', '=', False),
+            ('create_uid', '=', user.id)]</field>
+        <field name="perm_create" eval="False" />
+        <field name="perm_write" eval="False" />
+    </record>
+
+<!--    <record model="ir.rule" id="agent_own_orders_rule">-->
```
Comment (rousseldenis) [inline sale_commission_agent_restrict/security/security.xml]: Please remove unneeded entry

**Review by rousseldenis — CHANGES_REQUESTED** [review]
@ilyasProgrammer Thanks for this.

IMHO it's a little bit tricky. But at least solve my comments

**Comment by ilyasProgrammer** [comment]
@rousseldenis updated

**Comment by francesco-ooops** [comment]
@rousseldenis is it good now?

**Comment by francesco-ooops** [comment]
@pedrobaeza what do you say?

**Comment by francesco-ooops** [comment]
@OCA/crm-sales-marketing-maintainers please?

**Comment by elvise** [comment]
@OCA/crm-sales-marketing-maintainers please take a look 😃

**Comment by elvise** [comment]
@rousseldenis is it good now ? :)

**Comment by elvise** [comment]
@rousseldenis just a gentle reminder:)

File: sale_commission_agent_restrict/models/sale.py (hunk 2 above)
Comment (elvise) [reply sale_commission_agent_restrict/models/sale.py]: @rousseldenis now is it good ?

File: sale_commission_agent_restrict/security/security.xml (hunk 3 above)
Comment (elvise) [reply sale_commission_agent_restrict/security/security.xml]: @rousseldenis now is it good ?

**Review by rousseldenis — APPROVED** [review]

**Comment by francesco-ooops** [comment]
@rousseldenis merge too?

**Pull request merged**